models/DataObject/Service.php

```diff
@@ -1618,4 +1618,26 @@ public static function removeObjectFromSession($objectId)
     {
         self::removeElementFromSession('object', $objectId);
     }
+
+    /**
+     * @param array $objects
+     * @param string $sortBy
+     * @param string $childrenSortOrder
+     * @return array
+     */
+    public static function sortChildrenNaturally(array $objects, string $sortBy, string $childrenSortOrder): array
+    {
+        if('index' === $sortBy) {
+            $sortBy = 'idx';
+        }
+
+        usort(
+            $objects,
+            function ($a, $b) use ($sortBy) {
+                return strnatcasecmp($a[$sortBy], $b[$sortBy]);
```

Comment:
Unfortunately, this will not work with getChildren() as it compares based on array keys(array generated for `DataObjectController::treeGetChildsByIdAction`) so instead 

1. please change the `sortChildrenNaturally` definition to something like mentioned here: 
https://github.com/pimcore/pimcore/pull/5654#discussion_r373507502
which compares based on the actual object properties e.g. `$a->get('o_key')` 

2. and then you also have to move `DataObject\Service::sortChildrenNaturally` call to this line in treeGetChildsByIdAction
https://github.com/pimcore/pimcore/blob/ac773a11cb5d15c08f14160c2ca96991c37c2f02/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php#L182

Reply:
Sorry, I previously misread your comment. Plese check again. :)